connectors/es/sink.py

```diff
@@ -309,18 +313,27 @@ async def _run(self):
                         self._batch_bulk,
                         copy.copy(batch),
                         copy.copy(stats),
-                    )
+                    ),
+                    result_callback=functools.partial(self.batch_bulk_callback),
                 )
                 batch.clear()
                 stats = {OP_INDEX: {}, OP_UPSERT: {}, OP_DELETE: {}}
                 bulk_size = 0
 
             await asyncio.sleep(0)
 
-        await self.bulk_tasks.join()
+        await self.bulk_tasks.join(raise_on_error=True)
```

Comment:
We shouldn't wait for all the `bulk_tasks` to complete before we decide if there was an error. If any of them error, the sync should fail immediately. Right?

Reply:
Fail fast makes sense to me 👍

Reply:
Failing fast sounds good to me 👍

Reply:
+1 to that